Error compiling on OS X Mavericks

<!-- Welcome to the issues section if it's your first time! -->

### Before creating an issue, please ensure:
- [ ] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [ ] This issue is reproducible without changes to the code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1.Install all dependencies
2.CMake from build dir
3.Make

### Expected behavior
TFS compiles without any error

### Actual behavior
It shows this:

```
[  1%] Generating CXX prefix source cotire/tfs_CXX_prefix.cxx
[  2%] Generating CXX prefix header cotire/tfs_CXX_prefix.hxx
[  3%] Building CXX precompiled header cotire/tfs_CXX_prefix.hxx.pch
Scanning dependencies of target tfs
[  5%] Building CXX object CMakeFiles/tfs.dir/src/otpch.cpp.o
[  6%] Building CXX object CMakeFiles/tfs.dir/src/actions.cpp.o
In file included from /Users/XXX/forgottenserver/src/actions.cpp:1:
In file included from /Users/XXX/forgottenserver/build/cotire/tfs_CXX_prefix.hxx:4:
In file included from /Users/XXX/forgottenserver/build/cotire/tfs_CXX_prefix.cxx:4:
In file included from /Users/XXX/forgottenserver/src/otpch.h:28:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:213:16: error: 
      field has incomplete type 'value_type' (aka 'TalkAction')
    value_type __value_;
               ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:362:22: note: 
      in instantiation of template class
      'std::__1::__forward_list_node<TalkAction, void *>' requested here
    typedef typename __node::__self                                 __be...
                     ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:547:15: note: 
      in instantiation of template class
      'std::__1::__forward_list_base<TalkAction, std::__1::allocator<TalkAction>
      >' requested here
    : private __forward_list_base<_Tp, _Alloc>
              ^
/Users/XXX/forgottenserver/src/talkaction.h:54:33: note: in instantiation of
      template class 'std::__1::forward_list<TalkAction,
      std::__1::allocator<TalkAction> >' requested here
                std::forward_list<TalkAction> talkActions;
                                              ^
/Users/XXX/forgottenserver/src/talkaction.h:33:7: note: forward declaration
      of 'TalkAction'
class TalkAction;
      ^
In file included from /Users/XXX/forgottenserver/src/actions.cpp:1:
In file included from /Users/XXX/forgottenserver/build/cotire/tfs_CXX_prefix.hxx:4:
In file included from /Users/XXX/forgottenserver/build/cotire/tfs_CXX_prefix.cxx:4:
In file included from /Users/XXX/forgottenserver/src/otpch.h:28:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:550:28: error: 
      '__node_allocator' is a protected member of
      'std::__1::__forward_list_base<TalkAction, std::__1::allocator<TalkAction>
      >'
    typedef typename base::__node_allocator  __node_allocator;
                           ^
/Users/XXX/forgottenserver/src/talkaction.h:54:33: note: in instantiation of
      template class 'std::__1::forward_list<TalkAction,
      std::__1::allocator<TalkAction> >' requested here
                std::forward_list<TalkAction> talkActions;
                                              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:369:55: note: 
      declared protected here
                                                      __node_allocator;
                                                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:551:28: error: 
      '__node' is a protected member of
      'std::__1::__forward_list_base<TalkAction, std::__1::allocator<TalkAction>
      >'
    typedef typename base::__node            __node;
                           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:361:69: note: 
      declared protected here
    typedef __forward_list_node<value_type, void_pointer>           __node;
                                                                    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:552:28: error: 
      '__node_traits' is a protected member of
      'std::__1::__forward_list_base<TalkAction, std::__1::allocator<TalkAction>
      >'
    typedef typename base::__node_traits     __node_traits;
                           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:370:55: note: 
      declared protected here
    typedef allocator_traits<__node_allocator>        __node_traits;
                                                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:553:28: error: 
      '__node_pointer' is a protected member of
      'std::__1::__forward_list_base<TalkAction, std::__1::allocator<TalkAction>
      >'
    typedef typename base::__node_pointer    __node_pointer;
                           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:371:55: note: 
      declared protected here
    typedef typename __node_traits::pointer           __node_pointer;
                                                      ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:566:28: error: 
      'iterator' is a protected member of
      'std::__1::__forward_list_base<TalkAction, std::__1::allocator<TalkAction>
      >'
    typedef typename base::iterator       iterator;
                           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:401:65: note: 
      declared protected here
    typedef __forward_list_iterator<__node_pointer>             iterator;
                                                                ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:567:28: error: 
      'const_iterator' is a protected member of
      'std::__1::__forward_list_base<TalkAction, std::__1::allocator<TalkAction>
      >'
    typedef typename base::const_iterator const_iterator;
                           ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/forward_list:402:65: note: 
      declared protected here
    typedef __forward_list_const_iterator<__node_pointer>       const_iterator;
                                                                ^
7 errors generated.
make[2]: *** [CMakeFiles/tfs.dir/src/actions.cpp.o] Error 1
make[1]: *** [CMakeFiles/tfs.dir/all] Error 2
make: *** [all] Error 2
```

### Environment
Using an iMac late 2013 with OS X Mavericks and latest Xcode command line tools

--- a/src/talkaction.h
+++ b/src/talkaction.h
@@ -30,32 +30,6 @@ enum TalkActionResult_t {
 	TALKACTION_FAILED,
 };
 
-class TalkAction;
-
-class TalkActions : public BaseEvents
-{
-	public:
-		TalkActions();
-		~TalkActions();
-
-		// non-copyable
-		TalkActions(const TalkActions&) = delete;
-		TalkActions& operator=(const TalkActions&) = delete;
-
-		TalkActionResult_t playerSaySpell(Player* player, SpeakClasses type, const std::string& words) const;
-
-	protected:
-		LuaScriptInterface& getScriptInterface() final;
-		std::string getScriptBaseName() const final;
-		Event* getEvent(const std::string& nodeName) final;
-		bool registerEvent(Event* event, const pugi::xml_node& node) final;
-		void clear() final;
-
-		std::forward_list<TalkAction> talkActions;
-
-		LuaScriptInterface scriptInterface;
-};
-
 class TalkAction : public Event
 {
 	public:
@@ -79,6 +53,30 @@ class TalkAction : public Event
 
 		std::string words;
 		char separator = '"';
+};
+
+class TalkActions : public BaseEvents
+{
+	public:
+		TalkActions();
+		~TalkActions();
+
+		// non-copyable
+		TalkActions(const TalkActions&) = delete;
+		TalkActions& operator=(const TalkActions&) = delete;
+
+		TalkActionResult_t playerSaySpell(Player* player, SpeakClasses type, const std::string& words) const;
+
+	protected:
+		LuaScriptInterface& getScriptInterface() final;
+		std::string getScriptBaseName() const final;
+		Event* getEvent(const std::string& nodeName) final;
+		bool registerEvent(Event* event, const pugi::xml_node& node) final;
+		void clear() final;
+
+		std::forward_list<TalkAction> talkActions;
+
+		LuaScriptInterface scriptInterface;
 };
 
 #endif

